src/main.py

```diff
@@ -42,7 +39,14 @@ def concretize():
                          Static_Con,
                          Has_To_Left_Con, Has_To_Right_Con, Has_Behind_Con, Has_In_Front_Con, Is_Close_To_Con, Is_Medium_Distance_From_Con, Is_Far_From_Con,
                          On_Region_Con,
-                         Behavior_Con, Does_Maneuver_Con, Danger_Con, Collision_Con])
+                         Behavior_Con, Does_Maneuver_Con, Danger_Con, Collision_Con]) 
+    
+    for param in spec.params:
+        args.__dict__[param.key] = param.value
```

Comment:
How much validity check would we want? Right now it doesn't check anything but just adds everything to `args`

Reply:
Instead of this, i would probably check if `param.key` i already in args, then i would overwrite it. This way, we ensure that all allowed parameters are also included in the CLI, which means that the CLI would be the single source of complete truth in terms of what is relevant (and allowed) regarding the parameters. This is just an idea, you might find some practicality issues with this. i leave this up to you.

Reply:
Will `args` always contain a value for each of all the allowed parameter (key)?